### PR TITLE
Add the loading of cube map and the rendering of skybox but broken

### PR DESCRIPTION
**Pull request by samcake**
- Here is a very simple version to load the CubeMap from an URL and create the gpu::Texture

- Connect the ZOne entity sky url to the corresponding cube map texture and assign it

- THe rendering is not good, there is something happening in the way the texture is displayed that is wrong, still investigating

- Started designing the future of the TextureStorage that would take care of the good loading instead of the code happening in TextureCache.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 351b92627a4b570727d1466b038f520f52956008 available at: https://gist.github.com/0caf894e202b6b7725de<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4798.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4798.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: deb303838d8e0e6fbbdfd5198c7db12e9e1cce3e available at: https://gist.github.com/0caf894e202b6b7725de<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4798.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4798.dmg

File: libraries/render-utils/src/TextureCache.cpp [hunk 1]
```diff
@@ -515,9 +522,18 @@ void NetworkTexture::setImage(const QImage& image, bool translucent, const QColo
             formatGPU = gpu::Element(gpu::VEC4, gpu::UINT8, (isLinearRGB ? gpu::RGBA : gpu::SRGBA));
             formatMip = gpu::Element(gpu::VEC4, gpu::UINT8, (isLinearRGB ? gpu::BGRA : gpu::SBGRA));
         }
-        _gpuTexture = gpu::TexturePointer(gpu::Texture::create2D(formatGPU, image.width(), image.height(), gpu::Sampler(gpu::Sampler::FILTER_MIN_MAG_MIP_LINEAR)));
-        _gpuTexture->assignStoredMip(0, formatMip, image.byteCount(), image.constBits());
-        _gpuTexture->autoGenerateMips(-1);
+        
+        if (_type == CUBE_TEXTURE) {
+            if (_height >= (6 * _width)) {
```
Comment (ZappoMan) [inline libraries/render-utils/src/TextureCache.cpp]: wont block merge, but please fix magic number

File: libraries/model/src/model/Skybox.cpp [hunk 2]
```diff
@@ -21,6 +21,7 @@ using namespace model;
 
 Skybox::Skybox() {
 
+/* // PLease create a default engineer skybox
```
Comment (ZappoMan) [inline libraries/model/src/model/Skybox.cpp]: wont block merge, but please remove dead code or add in debug #ifdef or similar

**Pull request merged**